src/message_dao.py

```diff
@@ -12,27 +12,137 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, Optional
+import json
+import os
+import subprocess
+import zlib
+
+from functools import lru_cache
+from typing import Dict, List, Optional
+
+from dotenv import load_dotenv
+from google.cloud.sql.connector import Connector
 from message import Message
 
+load_dotenv()
 EPOCH_HASH = 'ae9e7be4a03765456fe38287533e6446e8bbc93c'
 
 class MessageDao(object):
+    def __init__(self) -> None:
+        self._initialize_connection()
+        self._initialize_tables()
+
+    def _initialize_connection(self) -> None:
+        connector = Connector()
+        self.connection = connector.connect(
+            os.environ.get("HOST"),
+            "pymysql",
+            user = os.environ.get("USER"),
+            password = os.environ.get("PASSWORD")
+        )
+    
+    def _initialize_tables(self) -> None:
+        db_name = os.environ.get("DB")
+        if not db_name:
+            raise Exception("Missing environment variable for name of database.")
+        with self.connection.cursor() as cursor:
+            cursor.execute("CREATE DATABASE IF NOT EXISTS " + db_name)
+            self.connection.select_db(db_name)
+            # Mapping from message id to message
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS Messages"
+                "(message_id VARCHAR(255) NOT NULL,"
+                "subject VARCHAR(255) NOT NULL,"
+                "from_ VARCHAR(255) NOT NULL,"
+                "in_reply_to VARCHAR(255),"
+                "content BLOB NOT NULL,"
+                "archive_hash VARCHAR(255),"
+                "change_id VARCHAR(255),"
+                "PRIMARY KEY (message_id))"
+            )
+            # Mapping from name of state attribute to 
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS States"
```

Comment:
so we're using this table as a generic key-value store?
Can we add a comment explaining that in this file somewhere?

We can either do it here or near the top of the file.
I thinking something like

```
We use two tables, Messages and States.
Message stores the messages we've uploaded and States is a key-value store which tracks things like "last_hash", the last Lore git commit we've processed.
```

Reply:
Good idea. I added a docstring in the init of the DAO to explain the tables that are being created.